extensions/amp-analytics/0.1/vendors/contentsquare.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS-IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {jsonLiteral} from '../../../../src/json';
+
+const CONTENTSQUARE_CONFIG = jsonLiteral({
+  'vars': {
+    'version': 'amp-0.0.1',
+    'csid':
+      '$MATCH(COOKIE(_cs_id), `([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})`, 1)',
```

Comment:
Can I ask why we need to read and write to `_cs_id`, instead of just using the `CLIENT_ID(AMP_CID)`? When will the value be different?

Reply:
@zhouyx the `_cs_id` cookie has a complex format: like: `<uuid>.<number>.<timestamp>.<number>.<timestamp>` 
In the future the goal is to be able to write such a cookie from an AMP page. But since we can't use variable in cookies, we have to find and other way to do it.

Reply:
Thanks for clarifying. sounds good to me. 
Also fyi: cookie value does support variables if I get it correctly. That say you can have
```
'cookies': {
    '_cs_id': {
      'value': 'CLIENT_ID(AMP_CID)TIMESTAMP',
  }
}
```